tests/test_bgp_multipath_relax.py

```diff
@@ -0,0 +1,126 @@
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    
+    # Find all V4 bgp neighbors from minigraph
+    bgp_v4nei = {}
+    for item in mg_facts['minigraph_bgp']:
+        if ':' in item['addr']:
+            continue
+        bgp_v4nei[item['name']] = item['addr']
+    return bgp_v4nei
+
+def test_bgp_multipath_relax(testbed, duthost):
+
+    logger.info("Starting test_bgp_multipath_relax on topology {}".format(testbed['topo']['name']))
+    topo_config = testbed['topo']['properties']['configuration']
+
+    bgp_v4nei = get_bgp_v4_neighbors_from_minigraph(duthost, testbed)
+
+    logger.info("bgp_v4nei {}".format(bgp_v4nei))
+    
+    # get all t0 routers name which has vips defined
+    dut_t0_neigh = get_t0_neigh(testbed, topo_config)
+
+    # get t2 neighbors
+    dut_t2_neigh = get_t2_neigh(testbed)
+    
+    if not dut_t0_neigh:
+        pytest.fail("Didn't find multipath t0's")
+
+    vips_prefix = get_vips_prefix(dut_t0_neigh, topo_config)
+    
+    logger.info("vips_prefix = {}".format(vips_prefix))
+
+    # find all paths of the prefix for test 
+    vips_t0, vips_asn = get_vips_prefix_paths(dut_t0_neigh, vips_prefix, topo_config)
+    
+    logger.info("vips_t0: {}, vips_asn: {}".format(vips_t0, vips_asn))
+
+    if len(vips_t0) <= 1:
+        pytest.fail("Did not find preconfigured multipath for the vips prefix under test")
+    
+    # Get the route from the DUT for the prefix
+    bgp_route = duthost.bgp_route(prefix=vips_prefix)['ansible_facts']['bgp_route']
+    
+    logger.info("Bgp route from DUT for prefix {} is {}".format(vips_prefix,bgp_route))
+
+    # Verify found vips prefix entry in Sonic bgp routes
+    assert bgp_route[vips_prefix]['found'] ==True
+
+    # Verify total multipath match number of t0 with vips that has prefix for test
+    assert int(bgp_route[vips_prefix]['path_num']) == len(vips_t0)
+
+    
+    # verify vips asn in each path of installed BGP vips prefix
+    for asn in vips_asn:
+        for aspath in bgp_route[vips_prefix]['aspath']:
+            assert str(asn) in  aspath
+
+    # gather one t2 neighbor advertised routes to validate routes advertised to t2 are correct with relaxed multipath
+    bgp_route_neiadv = duthost.bgp_route(neighbor=bgp_v4nei[dut_t2_neigh[0]], direction="adv")['ansible_facts']['bgp_route_neiadv']
+    
+    logger.info("Bgp neighbor adv from DUT for neigh {} and prefix {} is {}".
+                format(bgp_v4nei[dut_t2_neigh[0]],
+                        vips_prefix,
+                       bgp_route_neiadv[vips_prefix]))
+
+    # Verify vips prefix in advertised routes to t2 neighbor
+    assert bgp_route_neiadv.has_key(vips_prefix)
+
+    # vips prefix path has only 2 hops
+    assert len(bgp_route_neiadv[vips_prefix]['aspath']) == 2
+
+    assert int(bgp_route_neiadv[vips_prefix]['aspath'][0]) in vips_t0 and \
+        int(bgp_route_neiadv[vips_prefix]['aspath'][1]) in vips_asn
+
+
+
+
+
+
```

Comment:
Could you please remove the extra new lines at the end of the script?

Reply:
fixed in the latest commit